databases/week3/mealsharing.sql

```diff
@@ -0,0 +1,179 @@
+DROP DATABASE hyf_socialmedia;
+CREATE DATABASE meal_sharing;
+USE meal_sharing;
+
+CREATE TABLE `meal` (
+ `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+ `title` varchar(255) NOT NULL,
+ `description` text NOT NULL,
+ `location` varchar(255) NOT NULL,
+ `when` datetime NOT NULL,
+ `max_reservations` int(10) NOT NULL,
+ `price` decimal NOT NULL,
+ `created_date` date NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+INSERT INTO meal 
+VALUES (1, 'Lasagna', 'beef, onion, carrot, tomato', 'WA', '2020-06-30', 5, 4.95, '2020-06-01'),
+	(2, 'Rice poriddge', 'rice, milk, cinamon, brown sugar', 'CA', '2020-06-20', 10, 2.95, '2020-05-31'),
+    (3, 'Wok noodle', 'chicken, onion, carrot, bell pepper', 'CO', '2020-06-10', 10, 3.95, '2020-05-30');
+    
+INSERT INTO meal 
+VALUES (5, 'Waffel', 'flour, milk, egg, butter, sugar', 'MI', '2020-06-27', 8, 2.95, '2020-06-05'),
+	(6, 'Pizza', 'dough, tomato, salami, cheese, onion', 'CI', '2020-06-11', 6, 5.95, '2020-05-29'),
+    (7, 'Curry and rice', 'chicken, onion, carrot, potato', 'UT', '2020-06-05', 8, 4.95, '2020-05-28'),
+    (8, 'Vegan sushi', 'avocado, cucumber, soy sauce, rice, seaweed', 'NY', '2020-06-18', 2, 5.95, '2020-05-025'),
+	(9, 'Chocolate cake', 'flour, chocolate, sugar, butter, egg', 'NJ', '2020-06-07', 6, 6.95, '2020-05-23'),
+    (10, 'Bread', 'flour, egg, salt, yest', 'DC', '2020-06-13', 6, 4.95, '2020-05-22');
+
+-- Meal queries
+
+-- Get all meals
+SELECT * FROM meal;
+
+-- Add a new meal
+INSERT INTO meal 
+VALUES (4, 
+		'tomato soup', 
+        'tomato, onion, carrot, bacon', 
+        'TX', 
+        '2020-06-5', 
+        6, 
+        1.95, 
+        '2020-05-25');
+
+-- Get a meal with any id, fx 1
+SELECT * FROM meal WHERE id = 3;
+
+-- Update a meal with any id, fx 1. Update any attribute 
+-- fx the title or multiple attributes
+UPDATE meal SET max_reservations = 10 WHERE id = 1;
+
+-- Delete a meal with any id, fx 1
+DELETE FROM meal WHERE id = 2;
+
+
+CREATE TABLE `reservation` (
+ `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+ `number_of_guests` int(10) NOT NULL,
+ `meal_id` int(10) unsigned NOT NULL,
+ `location` varchar(255) NOT NULL,
+ `created_date` datetime NOT NULL,
+ CONSTRAINT `fk_reservation_meal_id` FOREIGN KEY (`meal_id`) REFERENCES `meal` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```

Comment:
You might consider moving all the create table statements to the top, so they are defined together.